OneSignalSDK/unittest/src/test/java/com/test/onesignal/MainOneSignalClassRunner.java

```diff
@@ -755,12 +859,38 @@ public void shouldNotFireNotificationOpenAgainAfterAppRestart() throws Exception
    @Test
    public void testOpenFromNotificationWhenAppIsInBackground() throws Exception {
       OneSignal.init(blankActivity, "123456789", ONESIGNAL_APP_ID, getNotificationOpenedHandler());
+      threadAndTaskWait();
```

Comment:
This test should be split into two different tests. 1. testing a click in the foreground, and the other a click in the background.